### PR TITLE
Add support for waypoints and routes

### PR DESCRIPTION
**Pull request by s3cur3**
I've tried to mirror the existing structure here (adding a field to the `Gpx` struct, and new structs for each of the types).

I'm using it at work and it's handled all the test data we've thrown at it so far. 😄

### CODE REVIEW

File: lib/gpx_ex/parser.ex [hunk 1]
```diff
@@ -39,17 +49,57 @@ defmodule GpxEx.Parser do
...
+    el = xpath(xml, ~x"./#{element_type}")
 
-  defp optinal_string(maybe_string) when maybe_string == "",  do: nil
-  defp optinal_string(maybe_string),  do: maybe_string
+    if is_nil(el) do
+      nil
+    else
+      xpath(el, ~x"./text()"s)
+    end
+  end
```
Comment (s3cur3) [inline lib/gpx_ex/parser.ex:96]: I was getting an exception thrown when trying to query `./whatever/text()` when the element in question didn't exist, so I moved to checking for the element, _then_ asking for its text.

File: mix.exs [hunk 2]
```diff
@@ -21,7 +21,7 @@ defmodule GpxEx.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:sweet_xml, "~> 0.6.0"}
+      {:sweet_xml, "~> 0.7.0"}
```
Comment (s3cur3) [inline mix.exs:24]: For this library's purposes, I _think_ 0.6.6 that we were using previously is actually compatible with 0.7.3, so we could potentially specify a broader supported range here, albeit at the cost of additional testing. 0.6.x is incompatible with a lot of libraries in the wild, including `ExAws`.

**Comment by caspg** [comment]
Nice one. Thanks for this PR.

**Comment by s3cur3** [comment]
My pleasure, and thank you! 🙏

**Pull request merged**